### PR TITLE
chore: update cryptography to >= 42.0.5

### PR DESCRIPTION
**Pull request by nigzak**
### SUMMARY
cryptography has some high CVE findings which are fixed with newer version.
To fix this the min-version in setup.py should be updated to latest version (42.0.5)

CVE-2024-26130⁠ (CVSS 7.5)
CVE-2023-50782⁠ (CVSS 7.5)
CVE-2024-0727⁠ (CVSS 5.5)
CVE-2023-49083⁠ (CVSS 5.9)
(and some low)

### BEFORE/AFTER SCREENSHOTS OR ANIMATED GIF
image

### TESTING INSTRUCTIONS
Open final build software and check that cryptography 41.0.5 (or newer) is inside

### ADDITIONAL INFORMATION

- [ ] Has associated issue:
- [ ] Required feature flags:
- [ ] Changes UI
- [ ] Includes DB Migration (follow approval process in [SIP-59](https://github.com/apache/superset/issues/13351))
  - [ ] Migration is atomic, supports rollback & is backwards-compatible
  - [ ] Confirm DB migration upgrade and downgrade tested
  - [ ] Runtime estimates and downtime expectations provided
- [ ] Introduces new feature or API
- [ ] Removes existing feature or API

### CODE REVIEW

File: setup.py [hunk 1]
```diff
@@ -81,7 +81,7 @@ def get_git_sha() -> str:
         "croniter>=0.3.28",
         "cron-descriptor",
         # snowflake-connector-python as of 3.7.0 doesn't support >=42.* therefore lowering the min to 41.0.2
-        "cryptography>=41.0.2, <43.0.0",
+        "cryptography>=42.0.5, <43.0.0",
```
Comment (dpgaspar) [inline setup.py:84]: We need to keep this at `41.0.2` because there are extra dependencies that colide with it

Comment (nigzak) [reply setup.py:84]: you mean to roll back the changes and only add the 42.0.5 version in base.in instead for now - right?

**Comment by sadpandajoe** [comment]
> This is being done here: #27281 snowflake just updated and released. After 27281 is merged I think we can close this one

I just merged #27281 so you can rebase and bump if you need to.

**Comment by nigzak** [comment]
cryptography 42.0.4 also has no findings (CVE scan) and it is already in master
=> closing this pull request now because it is outdated

**Pull request closed without merging**